lib/fs.js

```diff
@@ -228,41 +243,43 @@ fs.existsSync = function(path) {
 };
 
 fs.readFile = function(path, options, callback_) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
+  return new Promise((resolve, reject) => {
+    var callback = maybeCallback(arguments[arguments.length - 1], resolve, reject);
```

Comment:
`options` are optional, so, `callback_` may be on `options` argument.

But this implementation [already existed](https://github.com/nodejs/node/pull/4975/files#diff-9a205ef7ee967ee32efee02e58b3482dL231). I just added the `resolve` and `reject` arguments.